File: src/main/java/com/shreyash/BankApplication/repository/ContactRepository.java
package com.shreyash.BankApplication.repository;

import com.shreyash.BankApplication.entity.Contact;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ContactRepository extends JpaRepository<Contact, Long> {


    Boolean existsByPhoneNumber(String phoneNumber);

    Contact findByPhoneNumber(String PhoneNumber);

    Boolean existsByPanCard(String pancardNumber);

    Contact findByPanCard(String pancardNumber);

}

File: src/main/java/com/shreyash/BankApplication/repository/ContactRepository.kt
package com.shreyash.BankApplication.repository

import com.shreyash.BankApplication.entity.Contact
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ContactRepository: JpaRepository<Contact,Long> {

    fun existsByPhoneNumber(phoneNumber: String?): Boolean?

    fun findByPhoneNumber(phoneNumber: String?): Contact?

    fun existsByPanCard(panCardNumber: String?): Boolean?

    fun findByPanCard(panCardNumber: String?): Contact?
}